### DISCUSSION TITLE
How to enable cursor-down for command prompt

### DISCUSSION THREAD

**Discussion by tf42src** [LazyVim/LazyVim#3368] [Q&A]
I find selecting an item from a list by tab(ing) through it a bit unintuitive. Is there a way to allow the use of "cursor-down" to go down in the list?

![Screenshot from 2024-05-30 18-07-20](https://github.com/LazyVim/LazyVim/assets/9712861/3a17d714-0ea8-41df-8ffa-9179a07629ff)

Additional question: is there a fuzzy search for the command prompt? E.g. typing `format` doesn't show any results, I have to kind of know already what I am looking for.

**Comment by folke** [comment]
The noice prompt **IS** the same as the regular cmdline, so all keybindings are the same as well.
You could add extra terminal keybindings

**Reply by tf42src** [reply]
~I don't fully understand. I don't have `LazyExtras` on my command line, but (lazy)nvim knows the command (and others which are displayed). Also my command line (fish shell) _allows_ me to use "cursor-down" to select another option.~ - ah you mean the vim command line `:`.

Which keybindings are you suggesting? Mapping "cursor-down" to "tab" ?

**Reply by dpetka2001** [reply]
You can try the following in your `/lua/config/keymaps.lua`
```lua
vim.keymap.set("c", "<Up>", "<C-p>")
vim.keymap.set("c", "<Down>", "<C-Z>")
```
But with this you lose the functionality of recalling history items for command line. But I guess if that's what you want you're ok with it.

**Reply by tf42src** [reply]
Didn't work for me unfortunately. I wouldn't want losing going through history, hence I tried the following: 

```
vim.keymap.set("c", "<C-Up>", "<C-p>")
vim.keymap.set("c", "<C-Down>", "<C-Z>")
```

but also without success.

`vim.keymap.del` first, also didn't make any change.

I figured out, I can use <Left> and <Right> to go through the options, that might be good enough.

**Reply by dpetka2001** [reply]
In which file did you put those keymaps? I tried them and they worked just fine for me. Just put them in `~/.config/nvim/lua/config/keymaps.lua`.
